### PR TITLE
[teleport-ent] Allow configuration of cluster name

### PR DESCRIPTION
**Pull request by Nuru**
## what
- Allow configuration of Teleport cluster name. Before was forced to be same as Teleport Proxy Domain Name
- Add installation checklist

## why
- In Trusted Cluster configuration, Teleport cluster names cannot contain dots.
- Documentation.

### CODE REVIEW

File: releases/values/teleport-ent/Install.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+INSTALL checklist
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]: Lets standardize around the `README.md` which is automatically displayed when using GitHub.

Create an `## Install Checklist` chapter heading with the following content....

File: releases/values/teleport-ent/Install.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+- [ ] Update Geodesic, install teleport package
+- [ ] Decide on domain name for Teleport proxy
+- [ ] (MASTER ONLY) Find or generate ACM SSL certificate valid for that domain name
+- [ ] (MASTER ONLY) Set TELEPORT_SSL_CERTIFICATE_ARN to arn of SSL Certificate in Chamber
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
- [ ] (MASTER ONLY) Set `TELEPORT_SSL_CERTIFICATE_ARN` to arn of SSL Certificate in Chamber
```

File: releases/values/teleport-ent/Install.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+- [ ] Set TELEPORT_PROXY_DOMAIN_NAME in chamber
+- [ ] (Optional) Set TELEPORT_AUTH_DOMAIN_NAME in chamber
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
- [ ] (Optional) Set TELEPORT_AUTH_DOMAIN_NAME in chamber
```
```suggestion
- [ ] (Optional) Set `TELEPORT_AUTH_DOMAIN_NAME` in chamber
```

File: releases/values/teleport-ent/Install.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+INSTALL checklist
+
+- [ ] Update Geodesic, install teleport package
+- [ ] Decide on domain name for Teleport proxy
+- [ ] (MASTER ONLY) Find or generate ACM SSL certificate valid for that domain name
+- [ ] (MASTER ONLY) Set TELEPORT_SSL_CERTIFICATE_ARN to arn of SSL Certificate in Chamber
+- [ ] Set TELEPORT_PROXY_DOMAIN_NAME in chamber
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
- [ ] Set `TELEPORT_PROXY_DOMAIN_NAME` in chamber
```

File: releases/values/teleport-ent/Install.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+- [ ] (Optional) Set TELEPORT_AUTH_DOMAIN_NAME in chamber
+- [ ] (Optional) Set TELEPORT_CLUSTER_NAME in chamber
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
- [ ] (Optional) Set `TELEPORT_CLUSTER_NAME` in chamber
```

File: releases/values/teleport-ent/Install.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+- [ ] Set TELEPORT_VERSION in chamber
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
- [ ] Set `TELEPORT_VERSION` in chamber
```

File: releases/values/teleport-ent/Install.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+- [ ] Set TELEPORT_LICENSE_PEM in chamber (download file from portal)
+- [ ] Update kops-aws-platform to enable external-dns to manage higher level names
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
- [ ] Update `kops-aws-platform` to enable `external-dns` to manage higher level names
```

File: releases/values/teleport-ent/Install.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+- [ ] Update Geodesic, install teleport package
+- [ ] Decide on domain name for Teleport proxy
+- [ ] (MASTER ONLY) Find or generate ACM SSL certificate valid for that domain name
+- [ ] (MASTER ONLY) Set TELEPORT_SSL_CERTIFICATE_ARN to arn of SSL Certificate in Chamber
+- [ ] Set TELEPORT_PROXY_DOMAIN_NAME in chamber
+- [ ] (Optional) Set TELEPORT_AUTH_DOMAIN_NAME in chamber
+- [ ] (Optional) Set TELEPORT_CLUSTER_NAME in chamber
+- [ ] Set TELEPORT_VERSION in chamber
+- [ ] Set TELEPORT_LICENSE_PEM in chamber (download file from portal)
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
- [ ] Set `TELEPORT_LICENSE_PEM` in chamber (download file from portal)
```

File: releases/values/teleport-ent/Install.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+	- [ ] permitted_nodes="masters" # Which nodes are allowed to assume Teleport role. With kiam, it is only master nodes.
+	- [ ] chamber_service="kops"    # Service under which parameters are stored in chamber
+- [ ] Apply Terraform plan to provision resources
+	- [ ] Remember to update environment from Chamber to pick up generated parameters
+- [ ] Update conf/helmfiles
+	- [ ] Add to helmfile.yaml "releases/teleport-ent-proxy.yaml", - "releases/teleport-ent-auth.yaml"
+	- [ ] Update helmfile.envrc (optional): add teleport comments
+- [ ] Configure SAML
+	- [ ] Generate SAML Entity Descriptor from your SAML provider (XML file)
+		- [ ] Use "https://${TELEPORT_PROXY_DOMAIN_NAME}/v1/webapi/saml/acs" as both ACS and Entity ID
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
		- [ ] Use `https://${TELEPORT_PROXY_DOMAIN_NAME}/v1/webapi/saml/acs` as both ACS and Entity ID
```

File: releases/values/teleport-ent/Install.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+- [ ] Add config/teleport to provision teleport
+	- [ ] terraform.envrc, .envrc, terraform.tfvars, Makefile.tasks
+	- [ ] region="us-west-2"
+	- [ ] cluster_name="us-west-2.cloudposse.co"  # Used by kops-metadata to derive IAM role names for Master and Node instances
+	- [ ] permitted_nodes="masters" # Which nodes are allowed to assume Teleport role. With kiam, it is only master nodes.
+	- [ ] chamber_service="kops"    # Service under which parameters are stored in chamber
+- [ ] Apply Terraform plan to provision resources
+	- [ ] Remember to update environment from Chamber to pick up generated parameters
+- [ ] Update conf/helmfiles
+	- [ ] Add to helmfile.yaml "releases/teleport-ent-proxy.yaml", - "releases/teleport-ent-auth.yaml"
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
	- [ ] Add to `helmfile.yaml` the `- "releases/teleport-ent-proxy.yaml"`, and `- "releases/teleport-ent-auth.yaml"`
```

File: releases/values/teleport-ent/Install.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+	- [ ] Update helmfile.envrc (optional): add teleport comments
```
Comment (osterman) [inline releases/values/teleport-ent/Install.md]:
```suggestion
	- [ ] Update `helmfile.envrc` (optional): add teleport comments
```

**Review by osterman — APPROVED** [review]

File: releases/values/teleport-ent/teleport-ent-auth.yaml.gotmpl [hunk 12]
```diff
@@ -123,7 +123,7 @@ config:
       #
       # IMPORTANT: if you change cluster_name, it will invalidate all generated
       # certificates and keys (may need to wipe out /var/lib/teleport directory)
-      cluster_name: "{{ requiredEnv "TELEPORT_PROXY_DOMAIN_NAME" }}"
+      cluster_name: "{{ coalesce (env "TELEPORT_CLUSTER_NAME") (env "STAGE") (requiredEnv "TELEPORT_PROXY_DOMAIN_NAME") }}"
```
Comment (aknysh) [inline releases/values/teleport-ent/teleport-ent-auth.yaml.gotmpl:126]: should `TELEPORT_PROXY_DOMAIN_NAME` be required now since we use `coalesce`?

**Review by aknysh — COMMENTED** [review]
see comments

File: releases/values/teleport-ent/teleport-ent-auth.yaml.gotmpl (hunk 12 above)
Comment (Nuru) [reply releases/values/teleport-ent/teleport-ent-auth.yaml.gotmpl:126]: Yes, it is required elsewhere, and one of those 3 env vars must be set, so I require that one.

**Pull request merged**